tests/unit/test_lif_cell_group.cpp

```diff
@@ -0,0 +1,225 @@
+#include "../gtest.h"
+#include <cell_group_factory.hpp>
+#include <fstream>
+#include <lif_cell_description.hpp>
+#include <lif_cell_group_mc.hpp>
+#include <load_balance.hpp>
+#include <model.hpp>
+#include <recipe.hpp>
+#include <rss_cell.hpp>
+#include <rss_cell_group.hpp>
+
+using namespace arb;
+// Simple ring network of lif neurons.
+class ring_recipe: public arb::recipe {
+public:
+    ring_recipe(cell_size_type n, float weight, float delay):
+    ncells_(n + 1), weight_(weight), delay_(delay)
+    {}
+
+    cell_size_type num_cells() const override {
+        return ncells_;
+    }
+
+    // LIF neurons have gid in range [0..ncells_-2] whereas fake cell is numbered with ncells_ - 1.
+    cell_kind get_cell_kind(cell_gid_type gid) const override {
+        if (gid < cell_gid_type(ncells_ - 1)) {
+            return cell_kind::lif_neuron;
+        }
+        return cell_kind::regular_spike_source;
+    }
+
+    std::vector<cell_connection> connections_on(cell_gid_type gid) const override {
+        if (gid == cell_gid_type(ncells_ - 1)) {
+            return {};
+        }
+        // In a ring, each cell has just one incoming connection.
+        std::vector<cell_connection> connections;
+        cell_member_type source{(gid + cell_gid_type(ncells_) - 2) % (ncells_ - 1), 0};
+        cell_member_type target{gid, 0};
+        cell_connection conn(source, target, weight_, delay_);
+        connections.push_back(conn);
+
+        // Connect fake cell (numbered ncells_-1) to the first cell (numbered 0).
+        if (gid == 0) {
+            cell_member_type source{cell_gid_type(ncells_ - 1), 0};
+            cell_member_type target{gid, 0};
+            cell_connection conn(source, target, weight_, delay_);
+            connections.push_back(conn);
+        }
+
+        return connections;
+    }
+
+    util::unique_any get_cell_description(cell_gid_type gid) const override {
+        if (gid < cell_gid_type(ncells_ - 1)) {
+            return lif_cell_description();
+        }
+        // Produces just a single spike at time 0ms.
```

Comment:
I would make the case that is handled inside the `if` statement return the `rss_cell`, because it is the special case.